include/kframework/evm.md

```diff
@@ -1135,81 +1135,61 @@ Operators that require access to the rest of the Ethereum network world-state ca
 
 ### Account Queries
 
-TODO: It's unclear what to do in the case of an account not existing for these operators.
-`BALANCE` is specified to push 0 in this case, but the others are not specified.
-For now, I assume that they instantiate an empty account and use the empty data.
-
 ```k
     syntax UnStackOp ::= "BALANCE"
  // ------------------------------
     rule <k> BALANCE ACCT => BAL ~> #push ... </k>
+         <activeAccounts> ACCTS </activeAccounts>
          <account>
            <acctID> ACCT </acctID>
            <balance> BAL </balance>
            ...
          </account>
+      requires ACCT in ACCTS
 
-    rule <k> BALANCE ACCT => 0 ~> #push ... </k>
-         <activeAccounts> ACCTS </activeAccounts>
-      requires notBool ACCT in ACCTS
+    rule <k> BALANCE ACCT => 0 ~> #push ... </k> [owise]
 
     syntax UnStackOp ::= "EXTCODESIZE"
  // ----------------------------------
     rule <k> EXTCODESIZE ACCT => lengthBytes(CODE) ~> #push ... </k>
+         <activeAccounts> ACCTS </activeAccounts>
          <account>
            <acctID> ACCT </acctID>
            <code> CODE </code>
            ...
          </account>
+      requires ACCT in ACCTS
 
-    rule <k> EXTCODESIZE ACCT => 0 ~> #push ... </k>
-         <activeAccounts> ACCTS </activeAccounts>
-      requires notBool ACCT in ACCTS
+    rule <k> EXTCODESIZE _ => 0 ~> #push ... </k> [owise]
 
     syntax UnStackOp ::= "EXTCODEHASH"
  // ----------------------------------
     rule <k> EXTCODEHASH ACCT => keccak(CODE) ~> #push ... </k>
+         <activeAccounts> ACCTS </activeAccounts>
          <account>
            <acctID> ACCT </acctID>
            <code> CODE:Bytes </code>
            <nonce> NONCE </nonce>
            <balance> BAL </balance>
            ...
          </account>
-      requires notBool #accountEmpty(CODE, NONCE, BAL)
+      requires ACCT in ACCTS andBool notBool #accountEmpty(CODE, NONCE, BAL)
 
-     rule <k> EXTCODEHASH ACCT => 0 ~> #push ... </k>
-         <account>
-           <acctID> ACCT </acctID>
-           <code> CODE </code>
-           <nonce> NONCE </nonce>
-           <balance> BAL </balance>
-           ...
-         </account>
-       requires #accountEmpty(CODE, NONCE, BAL)
-
-    rule <k> EXTCODEHASH ACCT => 0 ~> #push ... </k>
-         <activeAccounts> ACCTS </activeAccounts>
-      requires notBool ACCT in ACCTS
-```
+    rule <k> EXTCODEHASH ACCT => 0 ~> #push ... </k> [owise]
 
-TODO: What should happen in the case that the account doesn't exist with `EXTCODECOPY`?
-Should we pad zeros (for the copied "program")?
```

Comment:
Same as above, do we have confirmation that what we have is the correct behavior?

Reply:
Yep, should be fine.